src/Drive.php

```diff
@@ -673,37 +685,174 @@ public function invite($recipient, SharingRole $role, ?\DateTimeImmutable $expir
      */
     public function getRoles(): array
     {
-        if((version_compare($this->ocisVersion, '6.0.0', '<'))) {
+        if(version_compare($this->ocisVersion, self::MIN_OCIS_VERSION_DRIVE_INVITE, '<')) {
             throw new EndPointNotImplementedException(Ocis::ENDPOINT_NOT_IMPLEMENTED_ERROR_MESSAGE);
-        };
-        $guzzle = new Client(
-            Ocis::createGuzzleConfig($this->connectionConfig, $this->accessToken)
-        );
+        }
+        $apiRoles = $this->sendGetPermissionsRequest()->getAtLibreGraphPermissionsRolesAllowedValues();
+        if (empty($apiRoles)) {
+            throw new InvalidResponseException(
```

Comment:
good idea :+1: 
I thought about returning an empty array in that case, but really an exception is the correct way to do it as a drive should always have some permissions given to some user